nestkernel/node_collection.cpp

```diff
@@ -208,6 +208,56 @@ NodeCollection::create( const std::vector< index >& node_ids_vector )
   return NodeCollection::create_( node_ids_vector );
 }
 
+NodeCollectionPTR
+NodeCollection::create_test_collection( const std::vector< index >& node_ids )
+{
+
+  if ( node_ids.empty() )
+  {
+    return NodeCollectionPTR( new NodeCollectionPrimitive() );
+  }
+
+  index current_first = node_ids[ 0 ];
+  index current_last = current_first;
+
+  std::vector< NodeCollectionPrimitive > parts;
+
+  index old_node_id = current_first;
+  for ( auto node_id = ++( node_ids.begin() ); node_id != node_ids.end(); ++node_id )
+  {
+    if ( *node_id == old_node_id )
```

Comment:
This implies that ids are sorted and will fail if they are not. I would suggest to explicitly require (in documentation) that `node_ids` is sorted and then test this here by `if ( *node_id < old_node_id )`.